lib/core/utils.js

```diff
@@ -160,6 +160,34 @@ function eachAsync(arr, eachFn, callback) {
   }
 }
 
+function eachAsyncSeries(arr, eachFn, callback) {
+  arr = arr || [];
+
+  let idx = 0;
+  let awaiting = arr.length;
+  if (awaiting === 0) {
+    callback();
+    return;
+  }
+
+  eachFn(arr[idx], eachCallback);
```

Comment:
nit: It's my preferences this should be below the `eachCallback` not to violate [`no-use-before-define`](https://eslint.org/docs/rules/no-use-before-define). If we're not using that rule then it's ok.

Reply:
fixed